spinalcordtoolbox/scripts/sct_compute_compression.py

```diff
@@ -0,0 +1,669 @@
+    # -----------------------------------------------------------
+    img_seg = Image(fname_segmentation).change_orientation('RPI')
+    img_labels = Image(fname_labels).change_orientation('RPI')
+    # Check if compression labels are in the same dimensions than spinal cord segmentation
+    if check_if_shape_mismatch(img_labels, img_seg):
+        raise ValueError(f"Shape mismatch between compression labels [{img_labels.data.shape}] and segmentation [{img_seg.data.shape}]). "
+                         f"Please verify that your compression labels wwere done in the same space as your input segmentation.")
+    # Call sct_process_segmentation to get morphometrics perslice in native space
+    path, file_name, ext = extract_fname(get_absolute_path(arguments.i))
+    fname_metrics = os.path.join(path, file_name + '_metrics' + '.csv')
+    sct_process_segmentation.main(argv=['-i', fname_segmentation, '-vertfile', arguments.vertfile, '-perslice', '1', '-o', fname_metrics])
+    metric = 'MEAN(' + arguments.metric + ')'  # Adjust for csv file columns name
+    # Fetch metrics of subject
+    df_metrics = pd.read_csv(fname_metrics).astype({metric: float})
+    # Get vertebral level corresponding to the slice with the compression
+    slice_thickness = get_slice_thickness(img_labels)
+    slice_compressed = get_compressed_slice(img_labels, verbose)
+    compressed_levels_dict = get_verterbral_level_from_slice(slice_compressed, df_metrics)
+    # Initialize variables if normalization
+    if arguments.normalize:
+        fname_metrics_PAM50 = os.path.join(path, file_name + '_metrics_PAM50' + '.csv')
+        # Call sct_process_segmentation to get morphometrics perslice in PAM50 space
+        sct_process_segmentation.main(argv=['-i', fname_segmentation, '-vertfile', arguments.vertfile, '-normalize-PAM50', '1',
+                                      '-perslice', '1', '-o', fname_metrics_PAM50])
+        sex = arguments.sex
+        age = arguments.age
+
+        # Step 2. Load reference input files (PAM50, healthy controls)
+        # ------------------------------------------------------------
+        # Select healthy controls based on sex and/or age range
+        path_ref = os.path.join(__data_dir__, 'PAM50_normalized_metrics')
+        if sex or age:
+            if not os.path.isfile(get_absolute_path(os.path.join(path_ref, arguments.file_participants))):
+                raise FileNotFoundError('participants.tsv file must exists to select sex or age.')
+            else:
+                fname_partcipants = get_absolute_path(os.path.join(path_ref, arguments.file_participants))
+                list_HC = select_HC(fname_partcipants, sex, age)
+        else:
+            list_HC = None
+        if age:
+            age.sort()
+            if any(n < 0 for n in age):
+                parser.error('Age range needs to be positive, {} was specified'.format(age))
+            # Put age range in order
+            else:
+                age.sort()
```

Comment:
I think this `age.sort()` call is duplicated.

Reply:
https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4003/commits/1fd5f62373fe0a6d67c85b88ab639f69caf2d604